integration/integration_test.go

```diff
@@ -0,0 +1,64 @@
+package integration
+
+import (
+	"testing"
+
+	"sigs.k8s.io/krew/integration/krew"
+)
+
+const (
+	// validPlugin is a valid plugin with a small download size
+	validPlugin = "konfig"
+)
+
+func TestInstall(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	krewTest, cleanup := krew.NewKrewTest(t)
+	defer cleanup()
+
+	krewTest.WithIndex().Cmd("install", validPlugin).RunOrFail()
+}
+
+func TestUpdate(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	krewTest, cleanup := krew.NewKrewTest(t)
+	defer cleanup()
+
+	krewTest.WithIndex().Cmd("update").RunOrFail()
+
+	indexFiles, err := krewTest.TempDir().List("index")
+	if err != nil {
+		t.Error(err)
+	}
+
+	if len(indexFiles) == 0 {
```

Comment:
I think this is fundamentally testing the wrong thing (implementation detail) rather than behavior. 

If we test here that after Update, “krew search” will work, it’s a better user focused test. 

We have plenty of “unit tests” exercising the  paths etc. so this is redundant.

Reply:
Agreed.